src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -14,34 +19,75 @@
 import java.util.Set;
 
 import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.EditClientCommand;
 import seedu.address.logic.commands.EditCommand;
+import seedu.address.logic.commands.EditTransactionCommand;
 import seedu.address.logic.parser.exceptions.ParseException;
 import seedu.address.model.tag.Tag;
 
 /**
- * Parses input arguments and creates a new EditCommand object
+ * Parses input arguments and creates a new EditClientCommand object
  */
 public class EditCommandParser implements Parser<EditCommand> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the EditCommand
      * and returns an EditCommand object for execution.
-     * @throws ParseException if the user input does not conform the expected format
+     * @throws ParseException if the user input does not conform the expected format.
      */
     public EditCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, PREFIX_MODE, PREFIX_NAME,
+                        PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG,
+                        PREFIX_GOODS, PREFIX_QUANTITY, PREFIX_PRICE, PREFIX_DATE);
 
         Index index;
 
         try {
             index = ParserUtil.parseIndex(argMultimap.getPreamble());
         } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    EditClientCommand.MESSAGE_USAGE), pe);
         }
+        String mode = argMultimap.getValue(PREFIX_MODE).orElse("");
 
-        EditCommand.EditClientDescriptor editClientDescriptor = new EditCommand.EditClientDescriptor();
+        switch (mode) {
+        case "client":
+            return parseClientEditCommand(argMultimap, index);
+        case "transaction":
+            return parseTransactionEditCommand(argMultimap, index);
+        case "company":
+        //return new EditCompany(args, index);
```

Comment:
Not sure if this can be deleted? Or is there a need for this?

Reply:
Thanks for highlighting this! I will delete this switch case